### PR TITLE
[Improve] Estimated column reader memory to control segment cache

### PR DESCRIPTION
**Pull request by Lchangliang**
## Proposed changes

Issue Number: close #xxx

If a segment has many columns, it will use a lot more memory than a segment with few columns, but it is not observable at present. So we estimated column reader memory to to control the memory.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Lchangliang** [comment]
run buildall

**Comment by xinyiZzz** [comment]
image

```
auto* lru_handle = LRUCachePolicy::insert(
            key.encode(), &value, 1, value.segment->meta_mem_usage(), CachePriority::NORMAL);
```
change to
```
auto* lru_handle = LRUCachePolicy::insert(
            key.encode(), &value, value.segment->meta_mem_usage(), value.segment->meta_mem_usage(), CachePriority::NORMAL);
```

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 1]
```diff
@@ -408,6 +408,7 @@ Status Segment::_create_column_readers(const SegmentFooterPB& footer) {
         RETURN_IF_ERROR(ColumnReader::create(opts, footer.columns(iter->second), footer.num_rows(),
                                              _file_reader, &reader));
         _column_readers.emplace(column.unique_id(), std::move(reader));
+        _meta_mem_usage += config::estimated_mem_per_column_reader;
```
Comment (xinyiZzz) [inline be/src/olap/rowset/segment_v2/segment.cpp:411]: in past, `_meta_mem_usage` only tracking some indexes memory,
now, add column reader memory

will there be other memory in segment?

Comment (Lchangliang) [reply be/src/olap/rowset/segment_v2/segment.cpp:411]: It still has some unique_ptr/shared_ptr object and the object `SubcolumnColumnReaders`.

**Comment by Lchangliang** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

File: be/src/common/config.cpp [hunk 2]
```diff
@@ -1053,6 +1053,10 @@ DEFINE_mInt32(schema_cache_sweep_time_sec, "100");
 
 // max number of segment cache, default -1 for backward compatibility fd_number*2/5
 DEFINE_mInt32(segment_cache_capacity, "-1");
+DEFINE_mInt32(estimated_num_columns_per_segment, "30");
+DEFINE_mInt32(estimated_mem_per_column_reader, "1024");
+// The value is calculate by storage_page_cache_limit * index_page_cache_percentage
```
Comment (xinyiZzz) [inline be/src/common/config.cpp:1058]: delete this line

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**